src/apps/intermediate-list/intermediate-list.tsx

```diff
@@ -0,0 +1,112 @@
+import React, { FC, useCallback, useEffect, useState } from "react";
+import { useSelector } from "react-redux";
+import { Link } from "../../components/atoms/link";
+import { useGetFeesV2 } from "../../core/fbs/fbs";
+import { FeeV2 } from "../../core/fbs/model";
+import { ModalIdsProps, useModalButtonHandler } from "../../core/utils/modal";
+import { useText } from "../../core/utils/text";
+import { useUrls } from "../../core/utils/url";
+import FeeList from "./fee-list/fee-list";
+import FeeDetailsModal from "./modal/fee-details-modal";
+import MyPaymentOverviewModal from "./modal/my-payment-overview-modal";
+import FeeDetailsContent from "./stackable-fees/fee-details-content";
+import {
+  getFeeObjectByFaustId,
+  getFeesPostPaymentChangeDate,
+  getFeesPrePaymentChangeDate
+} from "./utils/intermediate-list-helper";
+
+const IntermedateList: FC = () => {
+  const t = useText();
+  const { open } = useModalButtonHandler();
+  const { modalIds } = useSelector((s: ModalIdsProps) => s.modal);
+  const { data: fbsFees = [] } = useGetFeesV2<FeeV2[]>();
+  const [itemsPrePaymentChange, setItemsPrePaymentChange] = useState<
+    FeeV2[] | null
+  >(null);
+  const [totalFeePrePaymentChange, setTotalFeePrePaymentChange] =
+    useState<number>(0);
+  const [itemsPostPaymentChange, setItemsPostPaymentChange] = useState<
+    FeeV2[] | null
+  >(null);
+  const [totalFeePostPaymentChange, setTotalFeePostPaymentChange] =
+    useState<number>(0);
+  const { viewFeesAndCompensationRatesUrl } = useUrls();
+  const [faustIdentifier, setFaustIdentifier] = useState("0");
+  const [feeDetailsData, setFeeDetailsData] = useState<FeeV2[]>();
+  const openDetailsModalClickEvent = useCallback(
+    (faustId: string) => {
+      setFaustIdentifier(faustId);
+      setFeeDetailsData(getFeeObjectByFaustId(fbsFees, faustId));
+      open(faustId || "");
+    },
+    [fbsFees, open]
+  );
+
+  useEffect(() => {
+    if (fbsFees) {
+      const feesPrePaymentChange = getFeesPrePaymentChangeDate(fbsFees).length;
+      if (feesPrePaymentChange > 0) {
+        setItemsPrePaymentChange(getFeesPrePaymentChangeDate(fbsFees));
+      }
+      const feesPostPaymentChange = getFeesPrePaymentChangeDate(fbsFees).length;
+      if (feesPostPaymentChange > 0) {
+        setItemsPostPaymentChange(getFeesPostPaymentChangeDate(fbsFees));
+      }
+    }
+  }, [fbsFees]);
+
+  useEffect(() => {
+    let totalFee = 0;
+    itemsPrePaymentChange?.forEach((item) => {
+      totalFee += item.amount;
+    });
+    setTotalFeePrePaymentChange(totalFee);
```

Comment:
I recommend using `reduce` for situations like this. I think it is more elegant and you can control the variable scope better.